test/test_unstructured_2d.jl

```diff
@@ -21,8 +21,8 @@ isdir(outdir) && rm(outdir, recursive=true)
 
   @trixi_testset "elixir_euler_free_stream.jl" begin
     @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_euler_free_stream.jl"),
-      l2   = [3.357431396258123e-14, 2.439943089578555e-13, 1.4655386790023588e-13, 4.670410488845425e-13],
-      linf = [1.0169198816356584e-11, 6.838458965763294e-11, 4.456759961080081e-11, 1.4207657272891083e-10],
+      l2   = [3.3937971107485363e-14, 2.447586447887882e-13, 1.4585205789296455e-13, 4.716993468962946e-13],
```

Comment:
Why did these values change?

Reply:
I changed 
```julia
curve_values[k, 1] = quad_corners_flipped[m1, 1] + 0.5 * (cheby_nodes[k] + 1.0) * ( quad_corners_flipped[m2, 1]
                                                                                  - quad_corners_flipped[m1, 1])
curve_values[k, 2] = quad_corners_flipped[m1, 2] + 0.5 * (cheby_nodes[k] + 1.0) * ( quad_corners_flipped[m2, 2]
                                                                                  - quad_corners_flipped[m1, 2])
```
to be 
```julia
curve_values[k, 1] = linear_interpolate(cheby_nodes[k], quad_corners_flipped[m1, 1], quad_corners_flipped[m2, 1])
curve_values[k, 2] = linear_interpolate(cheby_nodes[k], quad_corners_flipped[m1, 2], quad_corners_flipped[m2, 2])
```
where internally the function `linear_interpolate`looks like
```julia
linear_interpolate(s, left_value, right_value) = 0.5 * ((1 - s) * left_value + (1 + s) * right_value)
```
So, effectively, this rearranged some multiplications and additions/subtractions when evaluating these linear interpolations (which can cause different roundoff errors). In turn, these slightly different roundoff errors propagate from the computation of the mapping -> computation of the metric terms resulting in slightly different values when computing values of the free-stream preservation (and well-balancedness). This change was "unseen" by the other test cases (like convergence), but free-stream and well-balancedness tests are particularly sensitive to roundoff error changes because their reference errors should be on the same magnitude.

Reply:
Thanks for the clarification!